Covert MediaStoreRequestHandler to Kotlin

amilcar-andrade opened the pull request:
* Also, stop using Utils.checkNotNull in favor of Kotlin's checkNotNull function

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+    try {
+      val contentResolver = context.contentResolver
+      val requestUri = checkNotNull(request.uri, { "request.uri == null" })
+      val exifOrientation = getExifOrientation(requestUri)
```
```suggestion
      val exifOrientation = getExifOrientation(requestUri)

```

not sure why the auto-conversation all empty lines, but it's annoying!

amilcar-andrade replied:
Oh true, will add spaces

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+import com.squareup.picasso3.Picasso.LoadedFrom
+
+internal class MediaStoreRequestHandler(context: Context) : ContentStreamRequestHandler(context) {
+
```
```suggestion
```

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+      val requestUri = checkNotNull(request.uri, { "request.uri == null" })
+      val exifOrientation = getExifOrientation(requestUri)
+      val mimeType = contentResolver.getType(requestUri)
+      val isVideo = mimeType != null && mimeType.startsWith("video/")
```
```suggestion
      val isVideo = mimeType != null && mimeType.startsWith("video/")

```

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+    var signaledCallback = false
+    try {
+      val contentResolver = context.contentResolver
+      val requestUri = checkNotNull(request.uri, { "request.uri == null" })
```
i wonder what's the difference between this and `request.uri!!`?

amilcar-andrade replied:
I did not want to just do `!!` given you guys were using the `Utils` stuff here, I can change it to `!!` if you prefer

jrodbx replied:
Ah, the difference is NPE vs IllegalStateException:

<img width="777" alt="Screen Shot 2022-01-16 at 9 47 18 PM" src="https://user-images.githubusercontent.com/1868149/149700505-be3f165b-56f7-47e2-9f35-96279e40333f.png">

Let's leave it as is, can always change it later if we change our minds.

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+          callback.onSuccess(Result.Bitmap(bitmap, LoadedFrom.DISK, exifOrientation))
+          return
+        }
+        val id = ContentUris.parseId(requestUri)
```
```suggestion

        val id = ContentUris.parseId(requestUri)

```

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+        val id = ContentUris.parseId(requestUri)
+        val options = checkNotNull(createBitmapOptions(request), { "options == null" })
+        options.inJustDecodeBounds = true
+        calculateInSampleSize(
```
```suggestion

        calculateInSampleSize(
```

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+          request.targetWidth, request.targetHeight, picassoKind.width,
+          picassoKind.height, options, request
+        )
+        val bitmap: Bitmap? = if (isVideo) {
```
```suggestion
        val bitmap = if (isVideo) {
```

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+            options
+          )
+        }
+        if (bitmap != null) {
```
```suggestion

        if (bitmap != null) {
```

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+          return
+        }
+      }
+      val source = getSource(requestUri)
```
```suggestion

      val source = getSource(requestUri)
```

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+      cursor = contentResolver.query(uri, CONTENT_ORIENTATION, null, null, null)
+      if (cursor == null || !cursor.moveToFirst()) {
+        0
+      } else cursor.getInt(0)
```
```suggestion
      } else {
        cursor.getInt(0)
      }
```

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+  }
+
+  internal enum class PicassoKind(val androidKind: Int, val width: Int, val height: Int) {
+    MICRO(
+      MediaStore.Images.Thumbnails.MICRO_KIND,
+      96,
+      96
+    ),
+    MINI(MediaStore.Images.Thumbnails.MINI_KIND, 512, 384), FULL(
+      MediaStore.Images.Thumbnails.FULL_SCREEN_KIND, -1, -1
+    );
```
```suggestion
    MICRO(MediaStore.Images.Thumbnails.MICRO_KIND, 96, 96),
    MINI(MediaStore.Images.Thumbnails.MINI_KIND, 512, 384), 
    FULL(MediaStore.Images.Thumbnails.FULL_SCREEN_KIND, -1, -1);
```

geez, the auto-formatter really mangled this one...

jrodbx commented on picasso/src/main/java/com/squareup/picasso3/MediaStoreRequestHandler.kt:
```diff
@@ -0,0 +1,136 @@
+  }
+
+  companion object {
+    private val CONTENT_ORIENTATION = arrayOf(
+      MediaStore.Images.ImageColumns.ORIENTATION
+    )
```
```suggestion
    private val CONTENT_ORIENTATION = arrayOf(MediaStore.Images.ImageColumns.ORIENTATION)
```

amilcar-andrade commented:
@jrodbx I think I addressed all your comments, except the usage of `checkNotNull`, let me know if you just want me to use `!!` was not super clear from the comment.
